framework/src/main/java/org/checkerframework/framework/ajava/InsertAjavaAnnotations.java

```diff
@@ -0,0 +1,580 @@
+                        result.putAll(AnnotationFileParser.annosInPackage(element));
+                    }
+                }
+            } else {
+                @SuppressWarnings("signature" // importDecl is non-wildcard, so its name is
+                // @FullyQualifiedName
+                )
+                @FullyQualifiedName String imported = importDecl.getNameAsString();
+                final TypeElement importType = elements.getTypeElement(imported);
+                if (importType != null && importType.getKind() == ElementKind.ANNOTATION_TYPE) {
+                    TypeElement annoElt = elements.getTypeElement(imported);
+                    if (annoElt != null) {
+                        result.put(annoElt.getSimpleName().toString(), annoElt);
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Inserts all annotations from the ajava file read from the stream {@code annotationFile} into
+     * a Java file with contents {@code fileContents} and returns the result.
+     *
+     * @param annotationFile input stream for an ajava file for {@code fileContents}
+     * @param fileContents contents of a Java file to insert annotations into
+     * @return a modified {@code fileContents} with annotations from {@code annotationFile} inserted
+     */
+    public String insertAnnotations(InputStream annotationFile, String fileContents) {
+        CompilationUnit annotationCu = StaticJavaParser.parse(annotationFile);
+        CompilationUnit fileCu = StaticJavaParser.parse(fileContents);
+        BuildInsertionsVisitor insertionVisitor = new BuildInsertionsVisitor(fileContents);
+        annotationCu.accept(insertionVisitor, fileCu);
+        List<Insertion> insertions = insertionVisitor.insertions;
+        // Insert annotations in reverse order of position. Making an insertion changes the offset
+        // values of everything after the insertion, so making the insertions in reverse order of
+        // removes the need to recalculate positions.
+        insertions.sort(
+                (insertion1, insertion2) -> {
+                    int cmp = Integer.compare(insertion1.position, insertion2.position);
+                    // Annotations belonging on their own line should be inserted before other
+                    // annotations. For example, in
+                    //
+                    // @Pure
+                    // @Tainted String myMethod();
+                    //
+                    // both annotations should be inserted at the same position (the start of
+                    // "String"), but @Pure should always appear first.
+                    if (cmp == 0 && (insertion1.ownLine != insertion2.ownLine)) {
+                        if (insertion1.ownLine) {
+                            cmp = -1;
+                        } else {
+                            cmp = 1;
+                        }
+                    }
+
+                    return -cmp;
```

Comment:
I suggest abstracting out this function so that you can give it a name and simplify this code.